### PR TITLE
response time assertion functionality

### PR DESCRIPTION
**Pull request by tdronkers04**
Adding response time functionality to `testRunner` through the use of [axios interceptors.](https://axios-http.com/docs/interceptors). The request interceptor adds a header `request-startTime` to the test http request. The response interceptor calculates the difference between the epoch stored in the `request-startTime` header and the current time and sets a new header `request-duration`.

Adding `responseTime` case to the `checkAssertions` method of `TestConfiguration` class.

### CODE REVIEW

File: entities/TestConfiguration.js [hunk 1]
```diff
@@ -38,6 +48,28 @@ class TestConfiguration {
   checkStatusCode(actual, expected) {
     return actual === expected;
   }
+
+  checkResponseTime(actual, comparison, target) {
```
Comment (scogra17) [inline entities/TestConfiguration.js:52]: Nit for consistency: what do you think about changing `target` to `expected` on this line or keeping as is and changing `expected` to `target` for the `checkStatusCode` params?

File: entities/TestConfiguration.js [hunk 2]
```diff
@@ -22,6 +22,16 @@ class TestConfiguration {
             results.push({ assertionType: assertionType, expectedValue: property, success: result })
           })
           break;
+        case "response_time":
+          result = this.checkResponseTime(
+            response.headers['request-duration'], 
+            this.assertions.response_time.comparison,
```
Comment (scogra17) [inline entities/TestConfiguration.js:28]: Nit: what do you think about parameterizing `this.assertions.response_time.comparison` to `this.assertions[assertionType].comparison`? Maybe silly, but enables changing "response_time" to something else, e.g. "roundtrip_trip" or whatever, in just one place

File: entities/TestRunner.js [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+  return config
+});
+
+axios.interceptors.response.use((response) => {
+  const currentTime = new Date().getTime();
+  const startTime = response.config.headers['request-startTime'] 
+  response.headers['request-duration'] = currentTime - startTime;
+  return response
+});
+
```
Comment (scogra17) [inline entities/TestRunner.js:14]: very cool!

Comment (scogra17) [reply entities/TestRunner.js:14]: 👍 Looks awesome overall!! I added a few nit picky comments that you may disagree with -- definitely feel free to ignore

**Review by scogra17 — APPROVED** [review]

File: entities/TestConfiguration.js (hunk 1 above)
Comment (tdronkers04) [reply entities/TestConfiguration.js:52]: sticking with `target` as per Checkly

**Pull request merged**